Error during restore mentioning libman.config.json in use.

### Describe the bug

Error `System.IO.IOException: The process cannot access the file 'C:\Users\ericstj\.librarymanager\libman.config.json' because it is being used by another process.`  full stack below.

My best guess here is a concurrency issue with multiple projects using the same task on different nodes.  This was a clean machine that never saw this repo before, nor used LibMan at all.

```

C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018: The "Microsoft.Web.LibraryManager.Build.RestoreTask" task failed unexpectedly. [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018: System.TypeInitializationException: The type initializer for 'Microsoft.Web.LibraryManager.Cache.WebRequestHandler' threw an exception. [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:  ---> System.TypeInitializationException: The type initializer for 'Microsoft.Web.LibraryManager.Configuration.Settings' threw an exception. [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:  ---> System.IO.IOException: The process cannot access the file 'C:\Users\ericstj\.librarymanager\libman.config.json' because it is being used by another process. [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options) [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode) [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode) [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode) [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at System.IO.StreamReader.ValidateArgsAndOpenPath(String path, Encoding encoding, Int32 bufferSize) [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at System.IO.StreamReader..ctor(String path) [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Configuration.Settings..ctor() [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Configuration.Settings..cctor() [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    --- End of inner exception stack trace --- [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Configuration.Settings.get_DefaultSettings() [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Configuration.ProxySettings.get_Default() [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Cache.WebRequestHandler..cctor() [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    --- End of inner exception stack trace --- [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Cache.WebRequestHandler.get_Instance() [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Build.Contracts.Dependencies.Initialize() [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Build.Contracts.Dependencies..ctor(IHostInteraction hostInteraction, IEnumerable`1 assemblyPaths) [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Build.Contracts.Dependencies.FromTask(String workingDirectory,IEnumerable`1 assemblyPaths) [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Build.RestoreTask.Execute() [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)[C:\src\dotnet\eshoponcontainers\src\Services\Identity\Identity.API\Identity.API.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018: The "Microsoft.Web.LibraryManager.Build.RestoreTask" task failed unexpectedly. [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018: System.TypeInitializationException: The type initializer for 'Microsoft.Web.LibraryManager.Cache.WebRequestHandler' threw an exception. [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:  ---> System.TypeInitializationException: The type initializer for 'Microsoft.Web.LibraryManager.Configuration.Settings' threw an exception. [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:  ---> System.IO.IOException: The process cannot access the file 'C:\Users\ericstj\.librarymanager\libman.config.json' because it is being used by another process. [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options) [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode) [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode) [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode) [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at System.IO.StreamReader.ValidateArgsAndOpenPath(String path, Encoding encoding, Int32 bufferSize) [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at System.IO.StreamReader..ctor(String path) [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Configuration.Settings..ctor() [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Configuration.Settings..cctor() [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    --- End of inner exception stack trace --- [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Configuration.Settings.get_DefaultSettings() [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Configuration.ProxySettings.get_Default() [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Cache.WebRequestHandler..cctor() [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    --- End of inner exception stack trace --- [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Cache.WebRequestHandler.get_Instance() [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Build.Contracts.Dependencies.Initialize() [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Build.Contracts.Dependencies..ctor(IHostInteraction hostInteraction, IEnumerable`1 assemblyPaths) [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Build.Contracts.Dependencies.FromTask(String workingDirectory,IEnumerable`1 assemblyPaths) [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Web.LibraryManager.Build.RestoreTask.Execute() [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
C:\.tools\.nuget\packages\microsoft.web.librarymanager.build\2.1.175\build\Microsoft.Web.LibraryManager.Build.targets(35,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)[C:\src\dotnet\eshoponcontainers\src\Web\WebStatus\WebStatus.csproj]
```

### To Reproduce
I cloned and tried to build https://github.com/dotnet-architecture/eshoponcontainers on a new machine.

### Expected behavior
Restore completes successfully.

### Additional context
Building with `dotnet build` of the root sln in eshop.  Using 8.0 RC2 dotnet

--- a/src/LibraryManager/Configuration/Settings.cs
+++ b/src/LibraryManager/Configuration/Settings.cs
@@ -64,22 +64,16 @@ namespace Microsoft.Web.LibraryManager.Configuration
                 }
                 catch (JsonReaderException)
                 {
-                    // If there were any errors, we'll reset the file
+                    // If there were any errors, we'll take the defaults
+                    // TODO: log a warning
                 }
             }
 
             if (_rootObject is null)
             {
-                InitSettingsFile(ConfigFilePath);
+                _rootObject = new JObject(new JProperty("config", new JObject()));
+                _configObject = _rootObject["config"] as JObject;
             }
-        }
-
-        private void InitSettingsFile(string configFilePath)
-        {
-            _rootObject = new JObject(new JProperty("config", new JObject()));
-            _configObject = _rootObject["config"] as JObject;
-
-            SaveSettingsFile(configFilePath, _rootObject);
         }
 
         /// <inheritdoc />

--- a/test/LibraryManager.Test/Configuration/SettingsTest.cs
+++ b/test/LibraryManager.Test/Configuration/SettingsTest.cs
@@ -47,37 +47,28 @@ namespace Microsoft.Web.LibraryManager.Test.Configuration
         }
 
         [TestMethod]
-        public void Constructor_FileDoesNotExist_WhenInitialized_CreateNewFile()
+        public void Constructor_FileDoesNotExist()
         {
             if (File.Exists(TestFilePath))
             {
                 File.Delete(TestFilePath);
             }
 
-            _ = new TestSettings(TestFilePath);
+            var settings = new TestSettings(TestFilePath);
 
-            Assert.IsTrue(File.Exists(TestFilePath));
+            // verify settings does not throw, does not have a well-known value
+            Assert.IsFalse(settings.TryGetValue(Constants.HttpsProxy, out _));
         }
 
         [TestMethod]
-        public void Constructor_FileExistsButEmpty_WhenInitialized_PopulateFile()
+        public void Constructor_FileExistsButEmpty()
         {
             File.WriteAllText(TestFilePath, "");
 
-            _ = new TestSettings(TestFilePath);
+            var settings = new TestSettings(TestFilePath);
 
-            Assert.IsTrue(new FileInfo(TestFilePath).Length > 0);
-        }
-
-        [TestMethod]
-        public void Constructor_FileExistsAndIsValid_WhenInitialized_DoNotModify()
-        {
-            string fileText = @"{ ""foo"": {} }";
-            File.WriteAllText(TestFilePath, fileText);
-
-            _ = new TestSettings(TestFilePath);
-
-            Assert.AreEqual(fileText, File.ReadAllText(TestFilePath));
+            // verify settings does not throw, does not have a well-known value
+            Assert.IsFalse(settings.TryGetValue(Constants.HttpsProxy, out _));
         }
 
         [TestMethod]
@@ -86,14 +77,10 @@ namespace Microsoft.Web.LibraryManager.Test.Configuration
             string fileText = @"[]"; // valid JSON, but not an object
             File.WriteAllText(TestFilePath, fileText);
 
-            _ = new TestSettings(TestFilePath);
+            var settings = new TestSettings(TestFilePath);
 
-            string actual = File.ReadAllText(TestFilePath);
-            string expected = @"{
-  ""config"": {}
-}";
-
-            Assert.AreEqual(StringUtility.NormalizeNewLines(expected), StringUtility.NormalizeNewLines(actual));
+            // verify settings does not throw, does not have a well-known value
+            Assert.IsFalse(settings.TryGetValue(Constants.HttpsProxy, out _));
         }
 
         [TestMethod]
@@ -121,19 +108,7 @@ namespace Microsoft.Web.LibraryManager.Test.Configuration
         }
 
         [TestMethod]
-        public void TryGetValue_ValueExists_ReturnTrueWithValue()
-        {
-            string fileText = @"{ ""config"": { ""testKey"": ""testValue"" } }";
-            File.WriteAllText(TestFilePath, fileText);
-
-            bool success = new TestSettings(TestFilePath).TryGetValue("testKey", out string value);
-
-            Assert.IsTrue(success);
-            Assert.AreEqual("testValue", value);
-        }
-
-        [TestMethod]
-        public void TryGetValue_ValueExistsInDifferentCase_ReturnFalse()
+        public void TryGetValue_SettingsNameIsCaseSensitive()
         {
             string fileText = @"{ ""config"": { ""testKey"": ""testValue"" } }";
             File.WriteAllText(TestFilePath, fileText);
@@ -212,7 +187,7 @@ namespace Microsoft.Web.LibraryManager.Test.Configuration
         }
 
         [TestMethod]
-        public void SetEncyptedValue_ValueIsDifferentFromOriginal()
+        public void SetEncryptedValue_ValueIsDifferentFromOriginal()
         {
             var ut = new TestSettings(TestFilePath);
 

